packages/pigeon/lib/kotlin_generator.dart

```diff
@@ -214,8 +213,8 @@ void _writeHostApi(Indent indent, Api api, Root root) {
 
               indent.write('channel.setMessageHandler ');
               indent.scoped('{ $messageVarName, reply ->', '}', () {
-                indent.writeln('val wrapped = hashMapOf<String, Any?>()');
-
+                indent.writeln('val wrapped = mutableListOf<Any?>()');
```

Comment:
It makes me a bit nervous that we are using the `add` method to build the wrapped value.  This can create subtle bugs.  I'd prefer if we used explicit indexes assigned for the different locations.